src/main/java/org/commonjava/storage/pathmapped/core/PathDBOutputStream.java

```diff
@@ -45,8 +45,11 @@ public void write( int b ) throws IOException
     {
         try
         {
-            super.write( b );
-            size += 1;
+            synchronized ( this )
```

Comment:
We don't need sync because this outputstream object should only be given to one thread. If more than one threads get it, then that is a bug.

Reply:
This is just a protection. I don't think it will break anything(both logic and performance) if it is in a single-thread env.

Reply:
I'd suggest we remove it in this case. If we keep it, someone else reading it might start thinking wildly about a multi-thread scenario and this could be misleading during debugging or something.

Reply:
Ok, I'll change it. But Actually from my perspective I still think that it will not bring misleading but more robustness. This will let reader notice that this outputstream should keep thread-safe. 
And if we want to make this as an underlying lib which some others want to refer, we should always consider the thread-safe as a needed option because we don't know actually if it will be used in what situation(we can not control if it is running in a multi-thread env).